### ISSUE TITLE
Omitting port in server URI sets port to 0

### ISSUE THREAD

**Issue by mainnet-pat** [woodser/monero-ts#203]
If I create a `MoneroWalletFull` with server URI without port like `https://monerod.slvit.us`, the fetch requests are going to be look like `https://monerod.slvit.us:0/json_rpc` and fail.

To work around this I need to set server to `https://monerod.slvit.us:443` which is unnecessary verboe

**Comment by mainnet-pat** [comment]
This still persists. I have to explicitly specify `https://node.sethforprivacy.com:443` instead of just `https://node.sethforprivacy.com`

**Comment by woodser** [comment]
Yeah [the PR](https://github.com/monero-project/monero/pull/9438) to moner-project has not been merged yet, so the behavior is still consistent with monero-project.

We *could* apply it as a custom extension to monero-project for the libraries in the meantime, but that does come with some overhead to maintain, so it would be nice if they simply merge it. :)

**Comment by mainnet-pat** [comment]
Why not simply check it monero-ts if port is omitted in http(s) then explicitly add :80(:443)?

**Comment by woodser** [comment]
Should be fixed by https://github.com/monero-project/monero/pull/9438.
